src/common/Eval.res

```diff
@@ -0,0 +1,68 @@
+type evalResult = result<string, string>
+
+@val external importMetaUrl: string = "import.meta.url"
+
+type workerMessage = ResultMessage({forCode: string, result: evalResult})
+type appMessage = EvalMessage(string)
+let source = "EvalSource"
+
+module Config = {
+  // TODO: Worker should send two events: LogMessage and ExceptionMessage
+  type fromWorker = {source: string, payload: workerMessage}
+  type fromApp = {source: string, payload: appMessage}
+  let make = () => %raw(`new Worker(new URL("./EvalWorker.mjs", import.meta.url))`)
+}
+
+module EvalWorker = Worker.Make(Config)
+
+type state = Idle | Evaluating(string) | Evaluated(string) | Error(string)
+type action =
+  | Evaluate(string)
+  | Success({forCode: string, message: string})
+  | Fail({forCode: string, message: string})
+
+let workerMessageToAction = message =>
+  switch message {
+  | ResultMessage({forCode, result: Ok(message)}) => Success({forCode: forCode, message: message})
+  | ResultMessage({forCode, result: Error(message)}) => Fail({forCode: forCode, message: message})
+  }
+
+let reducer = (state, action) =>
+  switch (state, action) {
+  | (Idle, Evaluate(code)) => Evaluating(code)
+  | (Evaluating(code), Success({forCode, message})) if forCode === code => Evaluated(message)
+  | (Evaluating(code), Fail({forCode, message})) if forCode === code => Error(message)
+  | (Error(_), Evaluate(code)) => Evaluating(code)
+  | (Evaluated(_), Evaluate(code)) => Evaluating(code)
+  | _ => state
+  }
+
+let useEval = () => {
+  let (state, dispatch) = React.useReducer(reducer, Idle)
```

Comment:
Todo, this state machine needs an extra transition (LOG) and FAIL can be changed to EXCEPTION.

Current: 

<img src="https://user-images.githubusercontent.com/9257001/125497444-e19fa5bf-f936-46ef-97c5-6b2199b2c553.png" width="400" />

Required: 

<img src="https://user-images.githubusercontent.com/9257001/125497761-e7921661-eb41-48bb-8e34-a852be90ca3f.png" width="400" />